packages/core-database-postgres/lib/connection.js

```diff
@@ -361,8 +361,6 @@ module.exports = class PostgresConnection extends ConnectionInterface {
     if (block.transactions.length > 0) {
       this.asyncTransaction.push(this.db.transactions.create(block.transactions))
     }
-
-    await this.db.tx(t => t.batch(this.asyncTransaction))
```

Comment:
Why was this removed? The transaction will no longer be executed and no data be stored.

Reply:
this is for fast rebuild only... transaction is executed when calling saveblockcommit... this is related.

And the method is no longer async - which is not in line with the core-database interface. this should be a separate PR anyways. just fixed for the import speed purpose.. now

Reply:
JavaScript itself has no interfaces so the "fake interface" is not a guarantee as implementations will differ between libraries. We shouldn't and can't take those interfaces as guarantees as some libraries are async and some sync.

Reply:
ok, then will leave it as is. thx

Reply:
i think this should be left as is.
However if sync is not needed (ie waiting for blocks being saved), we can call it without await.

as for transactions, the tx should be saved as well. (only header is saved on blocks table)

Reply:
i mean remove await here https://github.com/ArkEcosystem/core/blob/7e88a572fd09fb8bb8edd21746c340e17de725ee/packages/core-blockchain/lib/blockchain.js#L273

Reply:
but imo this is not the bottleneck

Reply:
actually I call this for get blocks

https://github.com/ArkEcosystem/core/blob/194fd3e2f9e2685460d5e41b972cfbec7bfe44c4/packages/core-database-postgres/lib/connection.js#L536-L560